shell/components/SortableTable/filtering.js

```diff
@@ -150,6 +177,24 @@ function columnsToSearchField(columns) {
 
 const ipLike = /^[0-9a-f\.:]+$/i;
 
+function handleStringSearch(searchFields, searchTokens, row) {
+  for ( let j = 0 ; j < searchTokens.length ; j++ ) {
```

Comment:
I think you want to use [`.every()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every).

```ts
return searchTokens.every(token => {
  const expect =!token.substr(0, 1) === '!'; // could you please add comments to this, why is it needed?
  const cleanToken = expect ? token.substr(1) : token;

  return cleanToken && !(matches(searchFields, cleanToken, row) !== expect)
})
```

Reply:
Also it would be great to extend it for a more generic scope, instead of just `'!'`